package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config

```diff
@@ -22,27 +15,14 @@ uci:section('simple-tc', 'interface', 'mesh_vpn', {
 	ifname = 'mesh-vpn',
 	enabled = vpn_config.limit_enabled,
 	limit_egress = vpn_config.limit_egress,
+	limit_ingress = vpn_config.limit_ingress,
 })
 
-if vpn == 'fastd' then
-	uci:set('fastd', 'mesh_vpn', 'enabled', vpn_config.enabled)
-	uci:set('simple-tc', 'mesh_vpn', 'limit_ingress', vpn_config.limit_ingress)
-else
-	uci:set('fastd', 'mesh_vpn', 'enabled', false)
-end
-uci:save('fastd')
-
-if vpn == 'tunneldigger' then
-	uci:set('tunneldigger', 'mesh_vpn', 'enabled', vpn_config.enabled)
-
-	if vpn_config.limit_enabled then
-		uci:set('tunneldigger', 'mesh_vpn', 'limit_bw_down', vpn_config.limit_ingress)
-	else
-		uci:delete('tunneldigger', 'mesh_vpn', 'limit_bw_down')
-	end
+vpn.enable(vpn_config.enabled)
+if vpn_config.limit_enabled then
+	vpn.set_limit(vpn_config.limit_ingress, vpn_config.limit_egress)
 else
-	uci:set('tunneldigger', 'mesh_vpn', 'enabled', false)
+	vpn.set_limit(nil, nil)
 end
-uci:save('tunneldigger')
```

Comment:
You are removing the code that disables the unavailable VPNs, this is not correct.

Reply:
Hmm, I wrote the code this way so that even VPNs that are not installed anymore get disabled (for the case when for example `fastd` is installed, but `gluon-mesh-vpn-fastd` isn't).

What we really want is a generic facility that allows removed packages to clean up after themselves on firmware upgrades, that way we wouldn't need a full list of packages that could exist. I have a few ideas how to solve this...